src/header/types/accept.rs

```diff
@@ -0,0 +1,299 @@
+    {
+        let mut accept = LinkedHashSet::new();
+        let mut present = false;
+        for value in values {
+            let media = value.as_str().split(',');
+            for mtype in media {
+                let media_type = MediaType::try_from(syntax::trim_whitespace(mtype))?;
+                accept.insert(media_type);
+            }
+            present = true;
+        }
+        if present {
+            Ok(Some(Accept(accept)))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Converts the [`Accept`] type to raw header values.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::convert::TryFrom;
+    ///
+    /// use rtsp::header::map::TypedHeader;
+    /// use rtsp::header::types::accept::MediaType;
+    /// use rtsp::header::types::Accept;
+    /// use rtsp::header::value::HeaderValue;
+    /// use rtsp::header::types::accept::QualityParam;
+    /// use mime::*;
+    /// use std::str::FromStr;
+    ///
+    /// let typed_header = vec![
+    ///     MediaType::new(mime::STAR_STAR, None), 
+    ///     MediaType::new(Mime::from_str("video/*").unwrap(), 
+    ///     Some(QualityParam::new(0.5)))
+    /// ].into_iter().collect::<Accept>();
+    /// let expected_raw_headers = vec![
+    ///     vec![HeaderValue::try_from("*/*, video/* ;q=0.5").unwrap()],
+    ///     vec![HeaderValue::try_from("video/* ;q=0.5, */*").unwrap()],
+    /// ];
+    /// let mut raw_header = vec![];
+    /// typed_header.encode(&mut raw_header);
+    /// assert!(raw_header == expected_raw_headers[0] ||
+    ///         raw_header == expected_raw_headers[1]);
+    /// ```
+    fn encode<Target>(&self, values: &mut Target)
+    where
+        Target: Extend<HeaderValue>,
+    {
+        // Unsafe Justification
+        //
+        // Mime types only allow for US-ASCII which is also valid UTF-8
+        let value = self.iter().map(MediaType::to_string).join(", ");
+        values.extend(once(unsafe { HeaderValue::from_string_unchecked(value) }));
+    }
+
+    fn header_name() -> &'static HeaderName {
+        &HeaderName::Accept
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct MediaType {
+    m_type: Mime,
+    quality: Option<QualityParam>,
+}
+
+impl MediaType {
+    pub fn new(m_type: Mime, quality: Option<QualityParam>) -> Self {
+        MediaType { m_type, quality }
+    }
+}
+
+impl fmt::Display for MediaType {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
+        let m_type = self.m_type.to_string();
+        match &self.quality {
+            Some(quality) => write!(f, "{} ;{}", m_type, quality),
+            None => write!(f, "{}", m_type),
+        }
+    }
+}
+
+impl<'media_type> TryFrom<&'media_type [u8]> for MediaType {
+    type Error = AcceptError;
+
+    fn try_from(value: &'media_type [u8]) -> Result<Self, Self::Error> {
+        let mut split = value.splitn(2, |&element| element == b';');
+        match split.next() {
```

Comment:
Let's see if we can clean this up a bit so it's not nested so much.

Something like this:

```
let mut split = value.splitn(2, |&element| element == b';');
let mediatype = split.next().ok_or(AcceptError::InvalidSyntax)?;
let raw_utf8 = str::from_utf8(mediatype).map_err(|_| AcceptError::InvalidSyntax)?;
let mime = raw_utf8.parse().map_err(|_| AcceptError::InvalidSyntax)?;
let m_type = mime;

match split.next() {
    Some(quality) => {
        let quality_param = QualityParam::try_from(quality).map_err(|_| AcceptError::InvalidSyntax);
        let quality = Some(quality_param);
        return Ok(MediaType { m_type, quality });
    }
    None => return Ok(MediaType { m_type, quality: None })
}
```